# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../xorg-server/gcc45.patch
# Copyright (C) 2014 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

log.c:199:9: error: #pragma GCC diagnostic not allowed inside functions
log.c:201:9: warning: format not a string literal, argument types not checked
log.c:212:9: error: #pragma GCC diagnostic not allowed inside functions
log.c:214:17: warning: format not a string literal, argument types not checked

--- xorg-server-1.16.2/hw/xfree86/os-support/linux/lnx_init.c.vanilla	2014-12-15 18:07:31.000000000 +0100
+++ xorg-server-1.16.2/hw/xfree86/os-support/linux/lnx_init.c	2014-12-15 18:08:15.000000000 +0100
@@ -166,7 +166,9 @@
 
         i = 0;
         while (vcs[i] != NULL) {
+#if __GNUC__ > 4 || __GNUC__ == 4 && __GNUC_MINOR__ > 5
 #pragma GCC diagnostic ignored "-Wformat-nonliteral"
+#endif
             snprintf(vtname, sizeof(vtname), vcs[i], xf86Info.vtno);    /* /dev/tty1-64 */
             if ((xf86Info.consoleFd = open(vtname, O_RDWR | O_NDELAY, 0)) >= 0)
                 break;
--- xorg-server-1.16.2/os/log.c.vanilla	2014-12-15 17:37:22.000000000 +0100
+++ xorg-server-1.16.2/os/log.c	2014-12-15 17:40:05.000000000 +0100
@@ -195,7 +195,7 @@
     char *logFileName = NULL;
 
     if (fname && *fname) {
-#if __GNUC__ > 4 || __GNUC__ == 4 && __GNUC_MINOR__ > 2
+#if __GNUC__ > 4 || __GNUC__ == 4 && __GNUC_MINOR__ > 5
 #pragma GCC diagnostic ignored "-Wformat-nonliteral"
 #endif
         if (asprintf(&logFileName, fname, display) == -1)
@@ -208,7 +208,7 @@
                 char *suffix;
                 char *oldLog;
 
-#if __GNUC__ > 4 || __GNUC__ == 4 && __GNUC_MINOR__ > 2
+#if __GNUC__ > 4 || __GNUC__ == 4 && __GNUC_MINOR__ > 5
 #pragma GCC diagnostic ignored "-Wformat-nonliteral"
 #endif
                 if ((asprintf(&suffix, backup, display) == -1) ||
--- xorg-server-1.16.2/test/signal-logging.c.vanilla	2014-12-15 18:14:52.000000000 +0100
+++ xorg-server-1.16.2/test/signal-logging.c	2014-12-15 18:16:06.000000000 +0100
@@ -113,12 +113,12 @@
                          -1, -2.00, -0.6023, -1203.30
                         };
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Woverflow"
 static void
 number_formatting(void)
 {
     int i;
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Woverflow"
     long unsigned int unsigned_tests[] = { 0,/* Zero */
                                            5, /* Single digit number */
                                            12, /* Two digit decimal number */
@@ -141,7 +141,6 @@
                                 -0x15D027BF211B37A, /* Large > 32 bit number */
                                 -0x7FFFFFFFFFFFFFFF, /* Maximum 64-bit signed number */
     } ;
-#pragma GCC diagnostic pop
 
     for (i = 0; i < sizeof(unsigned_tests) / sizeof(unsigned_tests[0]); i++)
         assert(check_number_format_test(unsigned_tests[i]));
@@ -152,6 +151,7 @@
     for (i = 0; i < sizeof(float_tests) / sizeof(float_tests[0]); i++)
         assert(check_float_format_test(float_tests[i]));
 }
+#pragma GCC diagnostic pop
 
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wformat-security"
